### PR TITLE
Add display of expressions to circuit drawers

### LINKED ISSUE #10236: Add support for visualising `Expr` values in control flow
### What should we add?

Once we have `Expr`-typed values in the new-style control-flow ops, we will want improved methods to visualise these.  This might take the form of a textual representation of the conditions, with any data-flow dependencies represented in the same wire-drawing manner as now.

Depends on #10223, #10224, #10096, #10170.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Adds expressions to mpl and text drawers

### Details and comments

Close #10236

This PR adds displays of expressions to ControlFlowOps in the mpl and text circuit drawers. The expressions are parsed using the QASM3 parser, so they appear the same as they would in a QASM3 string output.

Some of the changes in #10804 are incorporated in this PR, so that PR should be merged first.
```
qr = QuantumRegister(3, "qr")
cr = ClassicalRegister(3, "cr")
cr1 = ClassicalRegister(3, "cr1")
cr2 = ClassicalRegister(3, "cr2")
cr3 = ClassicalRegister(3, "cr3")
circuit = QuantumCircuit(qr, cr, cr1, cr2, cr3)

circuit.h(0)
with circuit.if_test(expr.equal(expr.bit_and(cr1, expr.bit_and(cr2, cr3)), 3)):
    circuit.z(0)

circuit.draw()
```

image
```
       ┌───┐┌─────────────────────────────── ┌───┐ ───────┐ 
 qr_0: ┤ H ├┤ If-0 (cr1 & (cr2 & cr3)) == 3  ┤ Z ├  End-0 ├─
       └───┘└───────────────╥─────────────── └───┘ ───────┘ 
 qr_1: ─────────────────────╫───────────────────────────────
                            ║                               
 qr_2: ─────────────────────╫───────────────────────────────
                            ║                               
 cr: 3/═════════════════════╬═══════════════════════════════
                        ┌───╨────┐                          
cr1: 3/═════════════════╡ [expr] ╞══════════════════════════
                        ├───╨────┤                          
cr2: 3/═════════════════╡ [expr] ╞══════════════════════════
                        ├───╨────┤                          
cr3: 3/═════════════════╡ [expr] ╞══════════════════════════
                        └────────┘                          

```

### CODE REVIEW

**Comment by enavarro51** [comment]
@jakelishman This PR has now been merged with main and conflicts resolved. Should be ready to go.

File: qiskit/visualization/circuit/matplotlib.py [hunk 1]
```diff
@@ -1543,6 +1584,17 @@ def _flow_op_gate(self, node, node_data, glob_data):
                 clip_on=True,
                 zorder=PORDER_FLOW,
             )
+            self._ax.text(
+                xpos - x_shift - 0.08,
```
Comment (jakelishman) [inline qiskit/visualization/circuit/matplotlib.py]: Should this 0.08 be `spacer`?

**Review by jakelishman — COMMENTED** [review]
I pushed up a minor tweak to some bits of documentation to make things easier.  I had one (very minor) inline question about a magic number, and my other thing is: could we add a test or two to each of the text and MPL drawers that conditionals in nested control-flow scopes work correctly?

**Comment by enavarro51** [comment]
In straightening out the mpl spacing, I found that for loops and case statements with only 1 qarg were printing outside the box in mpl. Also I redid display of the for loop range to fit for 1 qarg cases. Added tests for all this plus the nested expr's in both text and mpl.

**Review by jakelishman — APPROVED** [review]
Wonderful, thank you very very much!

**Pull request merged**